quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs

```diff
@@ -492,262 +465,223 @@ impl FieldPath {
-                "Error when parsing field `{}`: `field_mappings` is the only valid parameter.",
-                self.name
-            )
+        Type::I64 => {
+            let numeric_options: QuickwitNumericOptions = serde_json::from_value(json)?;
+            Ok(FieldMappingType::I64(numeric_options, cardinality))
         }
-        if self.is_array() {
-            bail!(
-                "Error when parsing field `{}`: array of object is not supported.",
-                self.name
-            )
+        Type::F64 => {
+            let numeric_options: QuickwitNumericOptions = serde_json::from_value(json)?;
+            Ok(FieldMappingType::F64(numeric_options, cardinality))
         }
-        let field_mappings = self
-            .field_mappings
-            .iter()
-            .map(|entry| FieldMappingEntry::try_from(entry.clone()))
-            .collect::<Result<Vec<_>, _>>()?;
-        if field_mappings.is_empty() {
-            bail!(
-                "Error when parsing field `{}`: object type must have at least one field mapping.",
-                self.name
-            )
+        Type::Date => {
+            let numeric_options: QuickwitNumericOptions = serde_json::from_value(json)?;
+            Ok(FieldMappingType::Date(numeric_options, cardinality))
         }
-        Ok(FieldMappingType::Object(field_mappings))
+        Type::Facet => unimplemented!("Facet are not supported in quickwit yet."),
+        Type::Bytes => {
+            let numeric_options: QuickwitNumericOptions = serde_json::from_value(json)?;
+            if numeric_options.fast && cardinality == Cardinality::MultiValues {
+                bail!("fast field is not allowed for array<bytes>.");
+            }
+            Ok(FieldMappingType::Bytes(numeric_options, cardinality))
+        }
+        Type::Json => todo!(),
     }
+}
+
+impl TryFrom<FieldMappingEntryForSerialization> for FieldMappingEntry {
+    type Error = String;
+
+    fn try_from(value: FieldMappingEntryForSerialization) -> Result<Self, String> {
+        validate_field_mapping_name(&value.name).map_err(|err| err.to_string())?;
+        let quickwit_field_type =
+            QuickwitFieldType::parse_type_id(&value.type_id).ok_or_else(|| {
+                format!(
+                    "Field `{}` has an unknown type: `{}`.",
+                    &value.name, &value.type_id
+                )
+            })?;
+        let mapping_type = deserialize_mapping_type(
+            quickwit_field_type,
+            serde_json::Value::Object(value.field_mapping_json),
+        )
+        .map_err(|err| format!("Error while parsing field `{}`: {}", value.name, err))?;
+        Ok(FieldMappingEntry::new(value.name, mapping_type))
+    }
+}
 
-    fn int_options(&self) -> anyhow::Result<NumericOptions> {
-        self.check_no_text_options()?;
-        let mut options = NumericOptions::default();
-        if self.stored {
-            options = options.set_stored();
+fn extract_map_from_json(
```

Comment:
Could be named closer to [as_object](https://docs.serde.rs/serde_json/value/enum.Value.html#method.as_object)